### ISSUE TITLE
Result status "computation error" changes to status "complete" upon reboot/restart

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#599]
**Reported by Dagorath on 13 May 38293734 08:53 UTC**
This error was reported in [this thread](http://boinc.berkeley.edu/dev/forum_thread.php?id=2611) on BOINC dev forums.

If user reboots or restarts BOINC when the manager has tasks with status "computation error" in the task list, the status in the task list gets changed to just "complete" on reboot/restart.  It would be much more user friendly if, after reboot/restart, the status would still somehow indicate an error occurred.  Perhaps it could say something like "complete but with computation error"?

**Comment by romw** [comment]
**Commented by Richard Haselgrove on 8 Jan 38293854 01:46 UTC**
I believe the same issue could also give rise to a discrepancy in the web site report for the WU.

Sometimes you see
```
Outcome        Client state
Client error   Compute error
```
and sometimes it's
```
Outcome        Client state
Client error   Done
```
It's probably caused by the same thing, and it would be good to check for web reporting at the same time as testing the Manager fix.

**Comment by romw** [comment]
**Commented by Nicolas on 15 Feb 38293954 13:46 UTC**
The state is saved incorrectly to client_state.xml. This is only noticed on a restart, because the client usually doesn't re-read the file, just keeps all state in memory.

Since the problem is state saving, it's not surprising it affects '''everything''' (reporting, manager, etc).

**Comment by romw** [comment]
**Commented by davea on 13 Dec 38373686 16:53 UTC**
(In [15231]) - client: when parsing state file, don't change result states
    unless there's an inconsistency
    Fixes #621

**Comment by romw** [comment]
**Commented by romw on 10 May 38375675 16:53 UTC**
(In [15237])     - client: when parsing state file, don't change result states
        unless there's an inconsistency
        Fixes #621

    client/
        client_types.C
